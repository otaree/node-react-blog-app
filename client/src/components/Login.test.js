import React from 'react';
import  { shallow, mount } from 'enzyme';
import { Login } from './Login';

describe("Login", () => {
    const mockLogin = jest.fn();
    const mockPush = jest.fn();
    let props = {
        login: mockLogin,
        history: {
            push: mockPush
        },
        authError: {
            isError: false,
            value: ''
        }
    };
    let login = shallow(<Login {...props} />);

    it('renders correctly', () => {
        expect(login).toMatchSnapshot();
    });

    describe('when form fields value changes', () => {
        beforeEach(() => {
            login = shallow(<Login {...props} />);
        });

        it('should set email on input change', () => {
            const value = "test@test.com";
            login.find('input').at(0).simulate('change', {
                target: { value }
            });
            expect(login.state('email').value).toEqual(value);
        });

        it('should set password on input change', () => {
            const value = "password!";
            login.find('input').at(1).simulate('change', {
                target: { value }
            });
            expect(login.state('password').value).toEqual(value);
        });
    });

    describe('when authentication Error', () => {
        beforeEach(() => {
            props.authError.isError = true;
            props.authError.value = "Authentication Fail";
            login = shallow(<Login {...props}/>);
        });

        it('renders correctly', () => {
            expect(login).toMatchSnapshot();
        });
    });

    describe('when valid from submission', () => {
        beforeEach(() => {
            login = mount(<Login {...props} />);
        });

        it('should submit the form correctly', () => {
            const user = {
                email: 'test@test.com',
                password: "password!"
            };
            login.setState({ email: { value: user.email, error: false } });
            login.setState({ password: { value: user.password, error: false } });            
            login.find('form').simulate('submit', {
                preventDefault: () => {}
            });       
            login.setProps({ isAuthenticated: true });     
            expect(mockLogin).toHaveBeenCalledWith(user);
            expect(mockPush).toHaveBeenCalledWith('/');
        });

        it('should render error when invalid form submission', () => {
            const user = {
                email: 'tesest.com',
                password: "pord!"
            };
            login.setState({ email: { value: user.email, error: false } });
            login.setState({ password: { value: user.password, error: false } });            
            login.find('form').simulate('submit', {
                preventDefault: () => {}
            });       
            expect(login.find('.auth__error').length).toEqual(2);            
        });
    });
});