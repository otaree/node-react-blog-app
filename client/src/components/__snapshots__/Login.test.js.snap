// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    authError={
        Object {
            "isError": false,
            "value": "",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
    login={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <div>
                <label>
                        Email
                </label>
                <input
                        onChange={[Function]}
                        type="email"
                        value=""
                />
        </div>
        <div>
                <label>
                        Password
                </label>
                <input
                        onChange={[Function]}
                        type="password"
                        value=""
                />
        </div>
        <div>
                <input
                        type="submit"
                        value="Submit"
                />
        </div>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div>
            <label>
                        Email
            </label>
            <input
                        onChange={[Function]}
                        type="email"
                        value=""
            />
</div>,
          <div>
            <label>
                        Password
            </label>
            <input
                        onChange={[Function]}
                        type="password"
                        value=""
            />
</div>,
          <div>
            <input
                        type="submit"
                        value="Submit"
            />
</div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Email
</label>,
              <input
                onChange={[Function]}
                type="email"
                value=""
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Password
</label>,
              <input
                onChange={[Function]}
                type="password"
                value=""
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              type="submit"
              value="Submit"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <div>
                    <label>
                              Email
                    </label>
                    <input
                              onChange={[Function]}
                              type="email"
                              value=""
                    />
          </div>
          <div>
                    <label>
                              Password
                    </label>
                    <input
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
          </div>
          <div>
                    <input
                              type="submit"
                              value="Submit"
                    />
          </div>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <div>
              <label>
                            Email
              </label>
              <input
                            onChange={[Function]}
                            type="email"
                            value=""
              />
</div>,
            <div>
              <label>
                            Password
              </label>
              <input
                            onChange={[Function]}
                            type="password"
                            value=""
              />
</div>,
            <div>
              <input
                            type="submit"
                            value="Submit"
              />
</div>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Email
</label>,
                <input
                  onChange={[Function]}
                  type="email"
                  value=""
/>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Password
</label>,
                <input
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password",
                },
                "ref": null,
                "rendered": "Password",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                type="submit"
                value="Submit"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Login when authentication Error renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    authError={
        Object {
            "isError": true,
            "value": "Authentication Fail",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
    login={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <div>
                Authentication Fail
        </div>
        <div>
                <label>
                        Email
                </label>
                <input
                        onChange={[Function]}
                        type="email"
                        value=""
                />
        </div>
        <div>
                <label>
                        Password
                </label>
                <input
                        onChange={[Function]}
                        type="password"
                        value=""
                />
        </div>
        <div>
                <input
                        type="submit"
                        value="Submit"
                />
        </div>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Authentication Fail
</div>,
          <div>
            <label>
                        Email
            </label>
            <input
                        onChange={[Function]}
                        type="email"
                        value=""
            />
</div>,
          <div>
            <label>
                        Password
            </label>
            <input
                        onChange={[Function]}
                        type="password"
                        value=""
            />
</div>,
          <div>
            <input
                        type="submit"
                        value="Submit"
            />
</div>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Authentication Fail",
          },
          "ref": null,
          "rendered": "Authentication Fail",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Email
</label>,
              <input
                onChange={[Function]}
                type="email"
                value=""
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Password
</label>,
              <input
                onChange={[Function]}
                type="password"
                value=""
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              type="submit"
              value="Submit"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <div>
                    Authentication Fail
          </div>
          <div>
                    <label>
                              Email
                    </label>
                    <input
                              onChange={[Function]}
                              type="email"
                              value=""
                    />
          </div>
          <div>
                    <label>
                              Password
                    </label>
                    <input
                              onChange={[Function]}
                              type="password"
                              value=""
                    />
          </div>
          <div>
                    <input
                              type="submit"
                              value="Submit"
                    />
          </div>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Authentication Fail
</div>,
            <div>
              <label>
                            Email
              </label>
              <input
                            onChange={[Function]}
                            type="email"
                            value=""
              />
</div>,
            <div>
              <label>
                            Password
              </label>
              <input
                            onChange={[Function]}
                            type="password"
                            value=""
              />
</div>,
            <div>
              <input
                            type="submit"
                            value="Submit"
              />
</div>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Authentication Fail",
            },
            "ref": null,
            "rendered": "Authentication Fail",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Email
</label>,
                <input
                  onChange={[Function]}
                  type="email"
                  value=""
/>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Password
</label>,
                <input
                  onChange={[Function]}
                  type="password"
                  value=""
/>,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password",
                },
                "ref": null,
                "rendered": "Password",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              false,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                type="submit"
                value="Submit"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
